app/overrides/add_sale_price_to_variant_index.rb

```diff
@@ -0,0 +1,15 @@
+Deface::Override.new(virtual_path: 'spree/admin/variants/index',
+  name: 'add_sale_price_to_admin_variant_index',
+  insert_before: "th.actions",
+  text: "
+    <th>Sale Price</th>")
+
+Deface::Override.new(virtual_path: 'spree/admin/variants/index',
+  name: 'add_sale_price_to_admin_variants_index_table',
+  insert_before: "td.actions",
+  text: "
+  <% if variant.sale_price.present? %>
```

Comment:
@rishabhguptarishi use the method i asked you to create in variant_decorator for this as well.